plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeLocalConcurrentCreateTableTest.java

```diff
@@ -0,0 +1,201 @@
+
+            String selectString = " as ((select stg1.a as a, stg1.b as b, stg1.d as part from stg_test stg1, stg_test stg2 where stg1.d=stg2.d) " +
+                    "union all (select stg1.a as a, stg1.b as b, stg1.d as part from stg_test stg1, stg_test stg2 where stg1.d=stg2.d))";
+
+            // Execute concurrent CTAS operations
+            executor.invokeAll(ImmutableList.<Callable<Void>>builder()
+                            .add(() -> {
+                                barrier.await(10, SECONDS);
+                                try {
+                                    getQueryRunner().execute("CREATE TABLE " + tableName +
+                                            (partitioned ? " WITH (partitioned_by = ARRAY['part'])" : "") + selectString);
+                                }
+                                catch (Exception e) {
+                                    RuntimeException trinoException = getTrinoExceptionCause(e);
+                                    assertThat(trinoException).hasMessageContaining("Table already exists");
+                                }
+                                return null;
+                            })
+                            .add(() -> {
+                                barrier.await(10, SECONDS);
+                                try {
+                                    getQueryRunner().execute("CREATE TABLE " + tableName +
+                                            (partitioned ? " WITH (partitioned_by = ARRAY['part'])" : "") + selectString);
+                                }
+                                catch (Exception e) {
+                                    RuntimeException trinoException = getTrinoExceptionCause(e);
+                                    assertThat(trinoException).hasMessageContaining("Table already exists");
+                                }
+                                return null;
+                            })
+                            .add(() -> {
+                                barrier.await(10, SECONDS);
+                                try {
+                                    getQueryRunner().execute("CREATE TABLE " + tableName +
+                                            (partitioned ? " WITH (partitioned_by = ARRAY['part'])" : "") + selectString);
+                                }
+                                catch (Exception e) {
+                                    RuntimeException trinoException = getTrinoExceptionCause(e);
+                                    assertThat(trinoException).hasMessageContaining("Table already exists");
+                                }
+                                return null;
+                            })
+                            .build())
+                    .forEach(MoreFutures::getDone);
+
+            // Verify table exists and has one row
```

Comment:
We should check that catalog doesn't have any files from failed attempts

Reply:
you mean the failed table creations?

Reply:
yes

Reply:
added the test case to ensure the files contained are from successful query execution and only